optuna/importance/_fanova.py

```diff
@@ -0,0 +1,128 @@
+from collections import OrderedDict
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from optuna._experimental import experimental
+from optuna.distributions import BaseDistribution
+from optuna.distributions import CategoricalDistribution
+from optuna.distributions import DiscreteUniformDistribution
+from optuna.distributions import IntUniformDistribution
+from optuna.distributions import LogUniformDistribution
+from optuna.distributions import UniformDistribution
+from optuna.importance._base import BaseImportanceEvaluator
+from optuna.importance._base import get_distributions
+from optuna.importance._base import get_study_data
+from optuna.study import Study
+
+try:
+    from ConfigSpace import ConfigurationSpace
+    from ConfigSpace.hyperparameters import CategoricalHyperparameter
+    from ConfigSpace.hyperparameters import Hyperparameter
+    from ConfigSpace.hyperparameters import UniformFloatHyperparameter
+    from ConfigSpace.hyperparameters import UniformIntegerHyperparameter
+    from fanova import fANOVA
+
+    _available = True
+except ImportError as e:
+    ConfigurationSpace = None
+    Hyperparameter = None
+    fANOVA = None
+
+    _import_error = e
+    _available = False
+
+
+@experimental("1.3.0")
+class FanovaImportanceEvaluator(BaseImportanceEvaluator):
+    """fANOVA parameter importance evaluator.
+
+    .. note::
+
+        Requires the `fanova <https://github.com/automl/fanova>`_ Python package.
+
+    .. seealso::
+
+        `An Efficient Approach for Assessing Hyperparameter Importance
+        <http://proceedings.mlr.press/v32/hutter14.html>`_.
+    """
+
+    def __init__(self) -> None:
+        _check_fanova_availability()
+
+    def evaluate(self, study: Study, params: Optional[List[str]]) -> Dict[str, float]:
+        distributions = get_distributions(study, params)
```

Comment:
I thought of that as well. However, I was ambivalent where to put these possibly overlapping logic. It could be placed in `get_param_importance`, but also `BaseImportanceEvaluator`. For now, until we have more evaluators, I'd like to keep the `get_param_importance` wrapper as thin as possible to not scatter the logic of the importance evaluation. I left a TODO comment. Does this seem acceptable?

Reply:
Thanks for letting me know your thought. It's acceptable.